### ISSUE TITLE
Needs a method that allows us to embed an arbitrary hash with a given name in a given file -eoi-

### ISSUE THREAD

**Issue by jjl** [jjl/asset-pack#4]

**Comment by kentfredric** [comment]
And this was why I had most of that complexity in metadata generation :P. 

Because it did allow you to pass arbitrary values and have them become visible as package-scoped `our` variables. :P

I had just done that by default for all values in `$meta` using named aliases becuase it was simpler than having two mechanisms for communication the same thing.

This was exactly the sort of "top level" mechanics I was suggesting :)

**Comment by kentfredric** [comment]
And the other reason I had it?

Imagine you have two "our" variables.

Imagine you need them to share a reference.

-> Bang.

Hence, its much easier to go:

``` perl
our $shared = $ structure created by Data::Dumper;
our $USERVALUE = $shared->{USERVALUE};
our $OTHERVALUE = $shared->{OTHERVALUE};
```

Because by then, all the cross-references are sorted out.

**Comment by kentfredric** [comment]
Fixed in bffc32cfe7b032ef91b9bd5ecea0d6cfafa5a626
